pkgs/by-name/sh/shadershark/package.nix

```diff
@@ -0,0 +1,73 @@
+{ stdenv
+, lib
+, fetchhg
+, epoxy
+, xorg
+, libGLU
+, glm
+, pkg-config
+, imagemagick
+, makeWrapper
+, curl
+, nix-prefetch-hg
+, nix
+, installShellFiles
+, writeShellApplication
+, common-updater-scripts
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "shadershark";
+  version = "0.1";
+
+  src = fetchhg {
+    url = "https://hg.globalcode.info/graphics/shader-shark";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-AYZWfqMckPKgXNIX9kAAv1mvD3opLi5EUElFsigiF3c=";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    makeWrapper
+    installShellFiles
+  ];
+
+  buildInputs = [
+    epoxy
+    xorg.libX11
+    libGLU
+    glm
+    imagemagick
+  ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/{bin,share/shadershark}
+
+    install -m755 -D build/shader-shark $out/bin
+    cp -r shaders textures $out/share/shadershark
+
+    wrapProgram $out/bin/shader-shark \
+      --set SHADER_SHARK_DATA_DIR $out/share/shadershark
+
+    installShellCompletion --bash --name shader-shark.bash bash-completion.sh
+
+    runHook postInstall
+  '';
+
+  passthru.updateScript = (lib.getExe (writeShellApplication {
+    name = "shader-shark-update-script";
+    runtimeInputs = [ curl common-updater-scripts ];
+    text = ''
+      repo_url=${finalAttrs.src.url}
+    '' + (builtins.readFile ./update.sh);
+  }));
```

Comment:
Not sure if I like this approach, seems unnecessarily complex.

My suggestion is to instead:
- Use `passthru.updateScript = [ ./update.sh finalAttrs.src.url ]`, since the list is passed to the script as parameters: https://nixos.org/manual/nixpkgs/unstable/#var-passthru-updateScript
- In the `update.sh` script, set `repo_url=$1`
- Also make the `update.sh` script self contained (e.g.: use `nix-shell` as shebang)

This will make it easier to maintain in the future.

Reply:
Done